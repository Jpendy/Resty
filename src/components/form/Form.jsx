import React from 'react';
import PropTypes from 'prop-types';

const Form = (apiUrl, fetchBody, onChange, onSubmit) => (
  <>    
    <form onSubmit={onSubmit} >
      <input type='text' value={apiUrl} onChange={onChange}  ></input>
      <label> GET <input type='radio' value='GET' name='fetch-method' ></input> </label>
      <label> POST <input type='radio' value='POST' name='fetch-method' ></input> </label> 
      <label> PUT <input type='radio' value='PUT' name='fetch-method' ></input> </label> 
      <label> DELETE <input type='radio' value='DELETE' name='fetch-method' ></input> </label>
      <textarea value={fetchBody} onChange={onChange}></textarea>
      <button>Go</button> 
    </form>
    
  </>
);

Form.propTypes = {
  apiUrl: PropTypes.string.isRequired,
  fetchBody: PropTypes.string,
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired
};

export default Form;
