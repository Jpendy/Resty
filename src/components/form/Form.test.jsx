import React from 'react';
import { shallow } from 'enzyme';
import Form from './Form';

describe('Form component', () => {
  it('renders Form', () => {
    const wrapper = shallow(<Form apiUrl='url' fetchBody='body' onChange={() => {}} onSubmit={() => {}}  />);
    expect(wrapper).toMatchSnapshot();
  });
});
