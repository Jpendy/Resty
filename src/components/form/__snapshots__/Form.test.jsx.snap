// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form component renders Form 1`] = `
<Fragment>
  <form
    onSubmit={[Function]}
  >
    <input
      name="apiUrl"
      onChange={[Function]}
      placeholder="API Url"
      type="text"
      value="url"
    />
    <div>
      <label>
        <input
          className="radio"
          name="fetchMethod"
          onChange={[Function]}
          type="radio"
          value="GET"
        />
         
        <p>
          GET
        </p>
         
      </label>
      <label>
        <input
          className="radio"
          name="fetchMethod"
          onChange={[Function]}
          type="radio"
          value="POST"
        />
         
        <p>
          POST
        </p>
         
      </label>
      <label>
        <input
          className="radio"
          name="fetchMethod"
          onChange={[Function]}
          type="radio"
          value="PUT"
        />
         
        <p>
          PUT
        </p>
         
      </label>
      <label>
        <input
          className="radio"
          name="fetchMethod"
          onChange={[Function]}
          type="radio"
          value="PATCH"
        />
         
        <p>
          PATCH
        </p>
         
      </label>
      <label>
        <input
          className="radio"
          name="fetchMethod"
          onChange={[Function]}
          type="radio"
          value="DELETE"
        />
         
        <p>
          DELETE
        </p>
         
      </label>
    </div>
    <textarea
      name="fetchBody"
      onChange={[Function]}
      placeholder="Body"
      value="body"
    />
    <button>
      Go
    </button>
  </form>
</Fragment>
`;
